fs/filter/filter.go

```diff
@@ -491,12 +491,16 @@ func (f *Filter) DirContainsExcludeFile(ctx context.Context, fremote fs.Fs, remo
 	if len(f.Opt.ExcludeFile) > 0 {
 		for _, excludeFile := range f.Opt.ExcludeFile {
 			exists, err := fs.FileExists(ctx, fremote, path.Join(remote, excludeFile))
-			if err != nil {
-				return false, err
-			}
 			if exists {
 				return true, nil
 			}
+			if err == fs.ErrorIsDir {
```

Comment:
This looks like the right approach. The logic needs to go like this though otherwise it ignores errors on the exists check

```go
if err != nil {
    if errors.Is(err, fs.ErrorIsDir) && strings.HasSuffix(excludeFile, "/") {
      return true, nil
    }
    return false, err
}
if exists {
    return true, nil
}
```

Reply:
above errors out in case exclude file doesn't have a `/` but name corresponds to a dir.

Eg:
```
test2
├── node-project
│   └── lol
└── python-project
    └── .venv/
```

```
rclone ls /Users/neo/tmp/test2 --exclude-if-present '.venv' --exclude-if-present '.git'
```

```
2025/08/30 18:59:27 ERROR : error listing: directory not found
2025/08/30 18:59:27 NOTICE: Failed to ls with 2 errors: last error was: directory not found
```

I wonder if we should even check for `/` suffix, since, rclone just crashes if it encounters a ignore-file name that exists as a directory in the fs.
Anyone relying on this behaviour is unlikely.

Reply:
above errors out in case exclude file doesn't have a `/` but name corresponds to a dir.

Eg:
```
test2
├── node-project
│   └── lol
└── python-project
    └── .venv/
```

```
rclone ls /Users/neo/tmp/test2 --exclude-if-present '.venv' --exclude-if-present '.git'
```

```
2025/08/30 19:05:19 ERROR : error listing: is a directory not a file
2025/08/30 19:05:19 NOTICE: Failed to ls with 2 errors: last error was: is a directory not a file
```

I wonder if we should even check for `/` suffix, since, rclone just crashes if it encounters a ignore-file name that exists as a directory in the fs.
Anyone relying on this behaviour is unlikely.

Reply:
above errors out in case exclude file doesn't have a `/` but name corresponds to a dir.

Eg:
```
test2
├── node-project
│   └── lol
└── python-project
    └── .venv/
```

```
rclone ls /Users/neo/tmp/test2 --exclude-if-present '.venv' --exclude-if-present '.git'
```

```
2025/08/30 19:05:19 ERROR : error listing: is a directory not a file
2025/08/30 19:05:19 NOTICE: Failed to ls with 2 errors: last error was: is a directory not a file
```

I wonder if we should even check for `/` suffix, since, rclone just crashes if it encounters a ignore-file name that exists as a directory in the fs.
Anyone relying on this behaviour is unlikely.

Reply:
Ha, yes you are right. Existing rclone has a bug! I was trying to preserve the existing behaviour, but let's fix the bug

So that makes your original solution nearly right but I think we need to add errors.Is, so

```patch
-                       if err == fs.ErrorIsDir {
+                       if errors.Is(err, fs.ErrorIsDir) {
                                return strings.HasSuffix(excludeFile, "/"), nil
                        }
```

Can you do that then add tests and docs (in `docs/content/filtering.md` also, and the help for --exclude-if-present would need adjusting.

https://github.com/rclone/rclone/blob/9e200531b1490000656031c42fff4d95477e1b46/fs/filter/filter.go#L36

Maybe something like

"Exclude directories if file or dir/ is present"

Thanks

Reply:
given it was a bug, do we even require the trailing `/` for directories ? most of the other commands do not distinguish against file/dir in this manner, (say copy).

Reply:
Yes, I agree with the reasoning by @nielash. I'm a little occupied with my day job but I do intend to finish the rest of it soon.